autoPyTorch/api/time_series_forecasting.py

```diff
@@ -0,0 +1,530 @@
+                                    get_dataset_compression_mapping)
+from autoPyTorch.datasets.base_dataset import (BaseDataset,
+                                               BaseDatasetPropertiesType)
+from autoPyTorch.datasets.resampling_strategy import (CrossValTypes,
+                                                      HoldoutValTypes,
+                                                      ResamplingStrategies)
+from autoPyTorch.datasets.time_series_dataset import (
+    TimeSeriesForecastingDataset, TimeSeriesSequence)
+from autoPyTorch.pipeline.time_series_forecasting import \
+    TimeSeriesForecastingPipeline
+from autoPyTorch.utils.hyperparameter_search_space_update import \
+    HyperparameterSearchSpaceUpdates
+
+
+class TimeSeriesForecastingTask(BaseTask):
+    """
+    Time Series Forecasting API to the pipelines.
+    Args:
+        seed (int): seed to be used for reproducibility.
+        n_jobs (int), (default=1): number of consecutive processes to spawn.
+        logging_config (Optional[Dict]): specifies configuration
+            for logging, if None, it is loaded from the logging.yaml
+        ensemble_size (int), (default=50): Number of models added to the ensemble built by
+            Ensemble selection from libraries of models.
+            Models are drawn with replacement.
+        ensemble_nbest (int), (default=50): only consider the ensemble_nbest
+            models to build the ensemble
+        max_models_on_disc (int), (default=50): maximum number of models saved to disc.
+            Also, controls the size of the ensemble as any additional models will be deleted.
+            Must be greater than or equal to 1.
+        temporary_directory (str): folder to store configuration output and log file
+        output_directory (str): folder to store predictions for optional test set
+        delete_tmp_folder_after_terminate (bool): determines whether to delete the temporary directory,
+            when finished
+        include_components (Optional[Dict]): If None, all possible components are used.
+            Otherwise specifies set of components to use.
+        exclude_components (Optional[Dict]): If None, all possible components are used.
+            Otherwise specifies set of components not to use. Incompatible with include
+            components
+    """
+
+    def __init__(
+        self,
+        seed: int = 1,
+        n_jobs: int = 1,
+        logging_config: Optional[Dict] = None,
+        ensemble_size: int = 50,
+        ensemble_nbest: int = 50,
+        max_models_on_disc: int = 50,
+        temporary_directory: Optional[str] = None,
+        output_directory: Optional[str] = None,
+        delete_tmp_folder_after_terminate: bool = True,
+        delete_output_folder_after_terminate: bool = True,
+        include_components: Optional[Dict] = None,
+        exclude_components: Optional[Dict] = None,
+        resampling_strategy: Union[
+            CrossValTypes, HoldoutValTypes
```

Comment:
you can use [`ResamplingStrategies`](https://github.com/dengdifan/Auto-PyTorch/blob/4898ca5ef492b465ad1331df2b380b10be070087/autoPyTorch/datasets/resampling_strategy.py#L111) for specifying the type here.